shared/lsif/providers.ts

```diff
@@ -39,50 +37,94 @@ export function createProviders(logger: Logger): Providers {
 export function createGraphQLProviders(
     queryGraphQL: QueryGraphQLFn<any> = sgQueryGraphQL,
     getRangeFromWindow?: Promise<RangeWindowFactoryFn>
-): Providers {
+): CombinedProviders {
     return {
-        definition: asyncGeneratorFromPromise(definition(queryGraphQL, getRangeFromWindow)),
+        definitionAndHover: cachedDefinitionAndHover(queryGraphQL, getRangeFromWindow),
         references: references(queryGraphQL, getRangeFromWindow),
-        hover: asyncGeneratorFromPromise(hover(queryGraphQL, getRangeFromWindow)),
         documentHighlights: asyncGeneratorFromPromise(documentHighlights(queryGraphQL, getRangeFromWindow)),
     }
 }
 
 /** The time to delay between range queries and an explicit definition/reference/hover request. */
 const RANGE_RESOLUTION_DELAY = 25
```

Comment:
Nit: I'm assuming this is milliseconds, but it would be better to have explicit descriptors for the unit either in the variable name i.e. RANGE_RESOLUTION_DELAY_MS or RANGE_RESOLUTION_DELAY_SECS or in the comment section: "/** time to delay between range queries in milliseconds.."

Reply:
Taken care of in https://github.com/sourcegraph/code-intel-extensions/pull/433.